mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -414,6 +414,9 @@ void AllocTracker::incrementExternalCount(uintptr_t ptr) {
 }
 
 void AllocTracker::decrementExternalCount(uintptr_t ptr) {
+  if (!llvm::is_contained(map, ptr))
+    return;
+
   assert(llvm::is_contained(map, ptr) &&
```

Comment:
This assertion will always be true now. @parthchadha Do we have a unit test to repro this issue? May be there is a better fix?

Reply:
Very hard to construct a unit test, it fails 100% time in SAM2 example though.